lib/bloomfilter.js

```diff
@@ -4,34 +4,29 @@ var bitcore = require('bitcore');
 var BloomFilter = require('bloom-filter');
 var BufferReader = bitcore.encoding.BufferReader;
 var BufferWriter = bitcore.encoding.BufferWriter;
-var $ = bitcore.util.preconditions;
-
 
+/**
+ * A constructor for Bloom Filters
+ * @see https://github.com/bitpay/bloom-filter
+ * @param {Buffer} - payload
+ */
 BloomFilter.fromBuffer = function fromBuffer(payload) {
+  var obj = {};
   var parser = new BufferReader(payload);
-  var data = parser.readVarLengthBuffer();
-  $.checkState(data.length <= BloomFilter.MAX_BLOOM_FILTER_SIZE,
-    'Filter data must be <= MAX_BLOOM_FILTER_SIZE bytes');
-  var nHashFuncs = parser.readUInt32LE();
-  $.checkState(nHashFuncs <= BloomFilter.MAX_HASH_FUNCS,
-    'Filter nHashFuncs must be <= MAX_HASH_FUNCS');
-  var nTweak = parser.readUInt32LE();
-  var nFlags = parser.readUInt8();
-
-  var vData = [];
-  var dataParser = new BufferReader(data);
-  for(var i = 0; i < data.length; i++) {
-    vData.push(dataParser.readUInt8());
+  var length = parser.readUInt8();
+  obj.vData = [];
+  for(var i = 0; i < length; i++) {
+    obj.vData.push(parser.readUInt8());
   }
+  obj.nHashFuncs = parser.readUInt32LE();
+  obj.nTweak = parser.readUInt32LE();
+  obj.nFlags = parser.readUInt8();
+  return new BloomFilter(obj);
```

Comment:
Yes, I was going to suggest to use `.create` but seems like it's a slightly different use case. Spawned https://github.com/bitpay/bloom-filter/issues/1